### PR TITLE
feat(size and space): update spacing unit ramp

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Created an internal function to create spacing and sizing stops classes
Removed old spacing CSS variables, left LESS variables because they're being used in a lot of places on the library, we will need to migrate them to fully remove old spacing variables.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [ ] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: lib/build/less/utilities/internals.less [hunk 1]
```diff
@@ -476,4 +476,29 @@
...
+
+      body {
+        --@{stop-100}:       calc(((@BU / 2) / 2) / 2);
+        --@{stop-200}:       calc((@BU / 2) / 2);
+        --@{stop-300}:       calc(@BU / 2);
+        --@{stop-400}:       @BU;
+        --@{stop-500}:       calc(@BU * 2);
+        --@{stop-600}:       calc((@BU * 2) * 2);
+        --@{stop-700}:       calc(((@BU * 2) * 2) * 2);
+        --@{stop-800}:       calc((((@BU * 2) * 2) * 2) * 2);
```
Comment (braddialpad) [inline lib/build/less/utilities/internals.less]: Less has a [pow function](https://lesscss.org/functions/#math-functions-pow) we can use for this

File: lib/build/less/utilities/internals.less [hunk 2]
```diff
@@ -476,4 +476,29 @@
...
+      @stop-400: ~"@{prefix}-400@{suffix}";
+      @stop-500: ~"@{prefix}-500@{suffix}";
+      @stop-600: ~"@{prefix}-600@{suffix}";
+      @stop-700: ~"@{prefix}-700@{suffix}";
+      @stop-800: ~"@{prefix}-800@{suffix}";
+
+      body {
+        --@{stop-100}:       calc(((@BU / 2) / 2) / 2);
+        --@{stop-200}:       calc((@BU / 2) / 2);
+        --@{stop-300}:       calc(@BU / 2);
```
Comment (braddialpad) [inline lib/build/less/utilities/internals.less]: Can probably use some math function for this as well.

Comment (juliodialpad) [reply lib/build/less/utilities/internals.less]: Updated

**Review by braddialpad — APPROVED** [review]
looks good thanks!

File: lib/build/less/variables/spacing.less [hunk 3]
```diff
@@ -28,48 +28,8 @@
...
-    sun114: -(@su114);
-    sun128: -(@su128);
-}
-
-body {
-    //  Spacing units
-    each(@spacing-units, {
-        --@{key}: @value;
-    });
-}
```
Comment (francisrupert) [inline lib/build/less/variables/spacing.less:75]: @juliodialpad,

Eek, we're going to need to retain `su` variables until we know specifically how the new spacing variable will be applied. Sorry I missed this before.

**Pull request merged**